### DISCUSSION TITLE
MongoDB Document Embedding Support for Different Documents?

### DISCUSSION THREAD

**Discussion by indicozy** [prisma/prisma#13437] [Q&A]
Hi everyone! I'm enjoying prisma with MongoDB!
I'm now building LMS for programming which is made from learning blocks like Markdown, mini-Quiz or Automatic Code Assesment.

I want to use MongoDB Embedding to use different type of blocks in a single query:

### Pseudo-Schema
```
enum BlockType {
  Markdown
  Quiz
  CodeAssessment
}

model Block {
  id          String       @id @default(auto()) @map("_id") @db.ObjectId
  createdAt   DateTime     @default(now())
  updatedAt   DateTime     @updatedAt
  type        BlockType
  data        Json?    // Here lies data of the embedded document
  Lesson      Lesson?      @relation(fields: [lessonId], references: [id])
  lessonId    String?      @db.ObjectId
  submissions Submission[]
}
Model Markdown {
Text      String
}
Model Quiz {
Question    String
Variants      String[]
Answer       Int
}
// and so on...
```

Is it planned to make this available in the future and would you find this type of Embedded Schema suitable for your workflow?

**Comment by nurul3101** [comment]
Hey @indicozy 👋 
Just to confirm what you mean is that in your model `Block` the column `data` could either refer to Markdown, Quiz which could potentially be `type` of data. Is that Correct?

On a sidenote the ideal place to create Feature Requests is [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=)

**Reply by indicozy** [reply]
Yes, It would be great if we could embed different types of data. One of the solutions could be to make use of SQL-like schema which neglects the power of NoSQL:
```diff
model Block {
  id          String       @id @default(auto()) @map("_id") @db.ObjectId
  createdAt   DateTime     @default(now())
  updatedAt   DateTime     @updatedAt
  type        BlockType
+ Quiz       Quiz?
+  Markdown    Markdown?
  // so on... 
  Lesson      Lesson?      @relation(fields: [lessonId], references: [id])
  lessonId    String?      @db.ObjectId
  submissions Submission[]
}
+ @@unique(Quiz, Markdown) // here
```

I found it to be tedious when I have different data for different types of blocks. I also need to set some handler to set which also neglects NoSQL principles.

Note that I'm still learning NoSQL schemas and maybe there can be a better solution. For me, we could use full features of the NoSQL to improve the DX. As a workaround, I create and validate `data`'s JSON in the backend, which is also not suiting the best.

So, if prisma had feature of embedding different types of documents to a single column, would it be better than SQL-like schema I described above?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
